sdk/src/create_signer.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+#![deny(missing_docs)]
+
+//! The `create_signer` module provides a way to obtain a [`Signer`]
+//! instance for each signing format supported by this crate.
+
+use std::path::Path;
+
+use crate::{
+    error::{Error, Result},
+    openssl::{EcSigner, EdSigner, RsaSigner},
+    signer::ConfigurableSigner,
+    Signer,
+};
+
+/// Creates a [`Signer`] instance using signing certificate and private key
+/// as byte slices.
+///
+/// The signing certificate and private key are passed to the underlying
+/// C++ code, which copies them into its own storage.
+///
+/// # Arguments
+///
+/// * `signcert` - Signing certificate
+/// * `pkey` - Private key
+/// * `alg` - Format for signing. Must be one of the supported
+///    formats (`rs256`, `rs384`, `rs512`, `ps256`, `ps384`, `ps512`,
+///    `es256`, `es384`, `es512`, or `ed25519`).
+/// * `tsa_url` - Optional URL for a timestamp authority
+pub fn from_keys(
```

Comment:
Is there a reason this is gated on the file_io feature? It's ok since you can't sign anything without file_io for now, but  just wondering.